internal/ethapi/tracer.go

```diff
@@ -240,21 +232,20 @@ func NewJavascriptTracer(code string, timeout time.Duration) (*JavascriptTracer,
 		db:         db,
 		dbvalue:    db.toValue(vm),
 		err:        nil,
-		timeout:    timeout,
 	}, nil
 }
 
+// Stop terminates execution of any JavaScript
+func (jst *JavascriptTracer) Stop(err error) {
+	jst.err = err
+	jst.vm.Interrupt <- func() {
+		panic(err)
+	}
+}
```

Comment:
Hmm, right I didn't think of that. We still need to get rid of the assignment of jst.err. When I say 'racy' I mean 'can be accessed concurrently from more than one goroutine' and that's not acceptable.